### PR TITLE
Test for Common/domain/replication_queue: GetMessagesfromDLQ & AckLevel

### PR DESCRIPTION
**Pull request by abhishekj720**
Why?: 
Strengthening the test coverage for key components of the replication queue, specifically for retrieving messages from the DLQ and querying acknowledgment levels. 

What has changed?:

Refined the GetMessagesFromDLQ test cases to validate successful message retrievals from DLQ, along with handling of read failures, ensuring the function behaves as expected in various scenarios.
Expanded the testing for GetAckLevel to cover successful acknowledgment level retrievals and to check the function's robustness against potential data retrieval issues.

How did you test it?:
unit tests with mock objects

Potential risks:
No risk involved.

### CODE REVIEW

File: common/domain/replication_queue_test.go [hunk 1]
```diff
@@ -210,3 +214,111 @@ func TestUpdateAckLevel(t *testing.T) {
...
+			rq := NewReplicationQueue(mockQueue, "testCluster", nil, nil)
+			tt.setupMock(mockQueue)
+			got, err := rq.GetAckLevels(context.Background())
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+			ctrl.Finish()
```
Comment (taylanisikdemir) [inline common/domain/replication_queue_test.go:255]: this finish call is not needed.

File: common/domain/replication_queue_test.go [hunk 2]
```diff
@@ -210,3 +214,111 @@ func TestUpdateAckLevel(t *testing.T) {
...
+
+func mockEncodeReplicationTask(sourceTaskID int64) ([]byte, error) {
+	var buf bytes.Buffer
+	buf.WriteByte(preambleVersion0)
+	binary.Write(&buf, binary.BigEndian, sourceTaskID)
+	return buf.Bytes(), nil
+}
+
+func TestGetMessagesFromDLQ(t *testing.T) {
+
```
Comment (taylanisikdemir) [inline common/domain/replication_queue_test.go:268]: nit: remove empty line

File: common/domain/replication_queue_test.go [hunk 3]
```diff
@@ -210,3 +214,111 @@ func TestUpdateAckLevel(t *testing.T) {
...
+			replicationTasks, token, err := rq.GetMessagesFromDLQ(context.Background(), tt.firstID, tt.lastID, tt.pageSize, tt.pageToken)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Len(t, replicationTasks, 1, "Expected one replication task to be returned")
+				assert.Equal(t, []byte("nextToken"), token, "Expected token to match 'nextToken'")
+			}
+			ctrl.Finish()
```
Comment (taylanisikdemir) [inline common/domain/replication_queue_test.go:321]: same here

**Review by taylanisikdemir — APPROVED** [review]

File: common/domain/replication_queue_test.go (hunk 1 above)
Comment (abhishekj720) [reply common/domain/replication_queue_test.go:255]: don't we require ctrl.Finish() to ensures all expected interactions with mock objects were met in a test?

**Pull request merged**